File: 2019/01/02.py
def calculate_fuel_needed_for_fuel(fuel):
    if fuel > 0:
        return fuel + calculate_fuel_needed_for_fuel(fuel // 3 - 2)
    return 0

with open("input") as file:
    data = [int(line) for line in file]

total_fuel = sum([calculate_fuel_needed_for_fuel(mass // 3 - 2) for mass in data])

print(total_fuel)
assert total_fuel == 5083024

File: 2019/01/02.swift
//
//  02.swift
//
//
//  Created by Brittany Pinder on 2021-08-17.
//

import Foundation

let contents = try! String(contentsOfFile: "Input")
let modules = contents.split(separator:"\n")
                      .compactMap { Int($0) }

func calculateFuel(mass: Int) -> Int {
    let fuel = max(mass / 3 - 2, 0)
    return fuel + (fuel > 0 ? calculateFuel(mass: fuel) : 0)
}

let totalFuel = modules.map { calculateFuel(mass: $0) }
                       .reduce(0, +)

print(totalFuel)
